### PR TITLE
Fix integer overflow in lseq with large float arguments

### PR DESCRIPTION
**Pull request by bitwisecook**
## Summary

- Fixed integer overflow panics in the `lseq` command when handling large float arguments that clamp to i64 extremes
- Added overflow-safe arithmetic checks using Zig's `@subWithOverflow`, `@mulWithOverflow`, and `@addWithOverflow` builtins
- Gracefully returns empty lists instead of panicking when overflow is detected
- Added comprehensive test coverage for edge cases including the regression test for `lseq 1e50 [expr {1e50+1}]`

### Runtime (`runtime/zig/cmds/list.zig`)

The `eval_lseq` function now uses overflow-safe arithmetic operations in three critical places:

1. **End value calculation** (3-argument form): When computing `end_val = start_val + (cnt - 1) * step_val`, each operation is checked for overflow. If any overflow occurs, an empty list is returned instead of panicking.

2. **Span calculation**: The span computation (`end_val - start_val` or `start_val - end_val`) now uses overflow-safe subtraction to handle cases where start/end are at i64 extremes.

3. **Loop increment**: Both ascending and descending loops now check for overflow before incrementing the loop variable `i`, preventing panics when the final element is at i64_max and the next increment would overflow.

### Tests (`tests/test_wasm_execution.py`)

Added four new test cases:
- `test_lseq_basic_count`: Basic functionality with single argument
- `test_lseq_start_end`: Two-argument form with ascending range
- `test_lseq_decreasing`: Two-argument form with descending range
- `test_lseq_large_double_does_not_panic`: Regression test ensuring `lseq 1e50 [expr {1e50+1}]` doesn't trap

### Test Baselines

Updated `lseq.toml` baseline:
- Changed from `W0-runtime-error` (trap) to `W-mixed-fail` (23 failing tests)
- Now runs 134 tests with 111 passing (previously crashed before running any tests)
- Removed the crash section; trap_allowed set to false

The `mathop.toml` baseline also changed from runtime error to timeout, indicating the lseq fix allowed test execution to progress further.

## Validation

- Added unit tests covering basic functionality and the specific regression case
- Existing test suite now executes lseq tests without trapping (111/134 passing)
- The remaining 23 failures are expected semantic differences (not yet supporting arbitrary-precision arithmetic series)

https://claude.ai/code/session_011dZ6m2PjVWLB7hzgb8CRgb

### CODE REVIEW

File: runtime/zig/cmds/list.zig [hunk 1]
```diff
@@ -713,18 +723,40 @@ fn eval_lseq(words: []const i32) result_mod.InterpResult {
...
+    // would itself trip the integer-overflow safety panic.
     const max_count: i64 = 16 * 1024 * 1024; // 16 M elements
-    const span: i64 = if (step_val > 0) end_val - start_val else start_val - end_val;
+    const span_res = if (step_val > 0)
+        @subWithOverflow(end_val, start_val)
+    else
+        @subWithOverflow(start_val, end_val);
+    if (span_res[1] != 0) return result_mod.from_globals(obj_new_string(0, 0));
+    const span: i64 = span_res[0];
     if (span < 0 or @divTrunc(span, if (step_val > 0) step_val else -step_val) > max_count) {
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply runtime/zig/cmds/list.zig]: Good catch — fixed in bdc1750. ``-step_val`` is now computed with ``@subWithOverflow(0, step_val)`` so an ``i64_min`` step (e.g. clamped from ``-1e50``) bails to an empty list instead of panicking. Pinned by the new ``test_lseq_negative_huge_double_step_does_not_panic`` regression — uses ``lseq 1e50 1e50 -1e50`` so the span subtraction guard sees zero and the negation site is the actual panic source.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: runtime/zig/cmds/list.zig [hunk 2]
```diff
@@ -713,18 +723,40 @@ fn eval_lseq(words: []const i32) result_mod.InterpResult {
...
     if (step_val > 0) {
-        while (i <= end_val) : (i += step_val) acc = rt.tcl_list(acc, obj_new_int(i));
+        while (i <= end_val) {
+            acc = rt.tcl_list(acc, obj_new_int(i));
+            // Stop before the increment overflows — this can happen
+            // when ``end_val`` is at i64_max (clamped from a float)
+            // and the loop emits the final element.
+            const next = @addWithOverflow(i, step_val);
+            if (next[1] != 0) break;
+            i = next[0];
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply runtime/zig/cmds/list.zig:768]: Confirmed and fixed in bdc1750. ``valtypes/tcl_list.zig:182`` ``memcpy``s ``a``'s bytes into a freshly-allocated buffer and never releases the inputs, so each iteration leaked one accumulator + one element obj — up to ~32 M TclObjs at the 16 M cap. Loop body now holds the new accumulator in a local, ``tcl_obj_release``s the prior ``acc`` and the element, then advances.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: runtime/zig/cmds/list.zig [hunk 3]
```diff
@@ -713,18 +723,40 @@ fn eval_lseq(words: []const i32) result_mod.InterpResult {
...
+        }
     } else {
-        while (i >= end_val) : (i += step_val) acc = rt.tcl_list(acc, obj_new_int(i));
+        while (i >= end_val) {
+            acc = rt.tcl_list(acc, obj_new_int(i));
+            const next = @addWithOverflow(i, step_val);
+            if (next[1] != 0) break;
+            i = next[0];
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply runtime/zig/cmds/list.zig:779]: Same fix applied symmetrically to the descending loop in bdc1750.

---
_Generated by [Claude Code](https://claude.ai/code)_
